src/build_vehicle_gui.cpp

```diff
@@ -1192,7 +1193,10 @@ struct BuildVehicleWindow : Window {
 		this->descending_sort_order = _engine_sort_last_order[type];
 		this->show_hidden_engines   = _engine_sort_show_hidden_engines[type];
 
+		this->SetCargoFilterArray();
 		this->UpdateFilterByTile();
+		this->eng_list.ForceRebuild();
+		this->GenerateBuildList(); // generate the list, since we need it in the next line
```

Comment:
Is this comment still correct, since the `next line` has changed? 😛

Reply:
I think it does... IIRC I moved it here because CreateNestedTree somehow requires engine list to count hidden and calculate the width of the button.

Reply:
Ok, it's actually FinishInitNested that needed the list so I updated the comment